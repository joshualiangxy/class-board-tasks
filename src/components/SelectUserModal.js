import React from 'react';
import Modal from 'react-modal';
import UserList from './UserList';

const SelectUserModal = ({
  users,
  groupCompletedState,
  removeUser,
  addUser,
  isOpen,
  onRequestClose
}) => (
  <Modal
    isOpen={isOpen}
    contentLabel="Select User"
    onRequestClose={onRequestClose}
    closeTimeoutMS={200}
    appElement={document.getElementById('root')}
  >
    <h2>Select users</h2>
    <UserList
      users={users}
      removeUser={removeUser}
      addUser={addUser}
      groupCompletedState={groupCompletedState}
    />
    <button onClick={onRequestClose}>Done</button>
  </Modal>
);

export default SelectUserModal;
