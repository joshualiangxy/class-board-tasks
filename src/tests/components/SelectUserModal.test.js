import React from 'react';
import { shallow } from 'enzyme';
import SelectUserModal from '../../components/SelectUserModal';
import users from '../fixtures/groupUsers';

const wrapper = shallow(<SelectUserModal users={users} />);

describe('render', () => {
  it('should render SelectUserModal', () => expect(wrapper).toMatchSnapshot());
});
