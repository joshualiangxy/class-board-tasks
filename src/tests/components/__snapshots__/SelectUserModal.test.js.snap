// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render SelectUserModal 1`] = `
<Modal
  appElement={null}
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  closeTimeoutMS={200}
  contentLabel="Select User"
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h2>
    Select users
  </h2>
  <UserList
    users={
      Array [
        Object {
          "admin": true,
          "displayName": "Robert",
          "studentNum": "A0123456A",
          "uid": "testuid",
        },
        Object {
          "admin": false,
          "displayName": "Peter",
          "studentNum": "A0000000A",
          "uid": "testuid2",
        },
      ]
    }
  />
  <button>
    Done
  </button>
</Modal>
`;
